isso/ext/notifications.py

```diff
@@ -139,17 +139,42 @@ def format(self, thread, comment, parent_comment, recipient=None, admin=False):
 
         else:
             uri = self.public_endpoint + "/id/%i" % parent_comment["id"]
-            key = self.isso.sign(('unsubscribe', recipient))
+            key = self.isso.sign(('unsubscribe', recipient_comment["email"]))
 
-            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient) + "/" + key))
+            rv.write("Unsubscribe from this conversation: %s\n" % (uri + "/unsubscribe/" + quote(recipient_comment["email"]) + "/" + key))
 
         rv.seek(0)
         return rv.read()
 
+    def notify_subject(self, thread, comment, parent_comment=None, recipient_comment=None, admin=False):
+        if admin:
+            return self.isso.conf.get("mail", "subject_admin").format(
+                title=thread["title"],
+                replier=comment["author"] or "Anonymous")
```

Comment:
As mentioned earlier, perhaps use a constant for "Anonymous" (not a member variable, as you had earlier)

Reply:
Okay, since there's no such thing called constant in python, I might implement it by a function returning a certain string.

Reply:
For translation of `Anonymous` in the future I might still use a member variable. 'Cause in that case every time I call the function is a query from i18n inside isso, so in that case I think it will be better to assign a string to a member variable according to the language.